### ISSUE TITLE
DataToolbarFilter: categoryName label required to delete chips

### ISSUE THREAD

**Issue by dlabrecq** [patternfly/patternfly-react#3552]
The `categoryName` prop of `DataToolbarFilter` is used to display the category name in the chip. In the example below, the "Project" category (i.e., chip) is a localizable string.

<img width="398" alt="Screen Shot 2020-01-22 at 8 13 49 PM" src="https://user-images.githubusercontent.com/17481322/72948869-7e13de80-3d54-11ea-82c0-6fbd32449aa0.png">

When the chip is deleted, the same string is provided as the `type` variable to callback.

<img width="853" alt="Screen Shot 2020-01-22 at 8 15 31 PM" src="https://user-images.githubusercontent.com/17481322/72948854-6b99a500-3d54-11ea-9631-7aaf09dddbb0.png">

In the filterable table demo below, the `type` is converted using `toLowerCase()`. However, considering that  this is a localizable label, we cannot use that approach to store filter values an array.

<img width="1037" alt="Screen Shot 2020-01-22 at 8 26 54 PM" src="https://user-images.githubusercontent.com/17481322/72949251-c41d7200-3d55-11ea-8fc3-51f35bc31b4f.png">

I don't want the localizable "Project" label here, but some ID (e.g., 'project'). (All my filter and menu options are based on IDs.) Perhaps a `categoryID` prop would be helpful here?

**Comment by dlabrecq** [comment]
When filtering by tag keys, I want to customize the label to read "Tag key: storageclass", where `storageclass` is the ID I'm using to store selections. This localization string does not work well with the chip deletion.

![Screen Shot 2020-01-23 at 11 14 38 AM](https://user-images.githubusercontent.com/17481322/73002321-d937e680-3dd1-11ea-9860-fa3255b6243c.png)

In this case, each tag key has its own value selections. Using the same `DataToolbarFilter` for each tag key, ends up looking like this:

![Screen Shot 2020-01-23 at 11 14 11 AM](https://user-images.githubusercontent.com/17481322/73002629-582d1f00-3dd2-11ea-8277-e70f924bd93e.png)

It looks like we need a unique `DataToolbarFilter` instance for each tag key?

**Comment by dlabrecq** [comment]
This is now a blocker for Cost Management. 

I encountered another case where the `categoryName` label is "Account", but the ID needed to delete the filter is `subscription_id`. I cannot delete the filter using `"Account".toLowerCase()` as shown in the filterable table demo.

Note that I'm reusing the data toolbar on 4 pages, so I cannot easily hard code a workaround. My OpenShift details page uses `account` as an ID, while our Azure page uses `subscription_id`; although, both pages display "Account" as the `categoryName` label.

**Comment by dlabrecq** [comment]
I replaced the workarounds in Cost Management with the new `DataToolbarChipGroup` interface. This works to delete chips with localized strings, but it's not consistent when deleting from other objects (i.e., a select component with multiple checkboxes).

When deleting a chip, my `deleteChip` callback receives this, which is the expected format:

<img width="259" alt="Screen Shot 2020-04-20 at 4 12 38 PM" src="https://user-images.githubusercontent.com/17481322/79795340-c7f36d00-8321-11ea-8cbd-976e15b3131b.png">

However, when the user deselects an option from the select component, I receive this instead:

<img width="159" alt="Screen Shot 2020-04-20 at 4 03 09 PM" src="https://user-images.githubusercontent.com/17481322/79794565-78f90800-8320-11ea-95fb-c2ab8b421f7e.png">

Note that the select component has the same selections as the chips:

<img width="489" alt="Screen Shot 2020-04-20 at 4 08 57 PM" src="https://user-images.githubusercontent.com/17481322/79795004-43085380-8321-11ea-9556-9d60ea41a123.png">

When deleting, we don't necessarily need the localized string in the `deleteChip` callback. (I was able to add a workaround to work with the different data structures.) However, it would be ideal the same data structure was received for all cases.

<img width="699" alt="Screen Shot 2020-04-20 at 5 33 20 PM" src="https://user-images.githubusercontent.com/17481322/79801950-11958500-832d-11ea-9983-8762275bfc88.png">
